mission_control/navigator_alarm/navigator_alarm_handlers/thruster_fault.py

```diff
@@ -0,0 +1,50 @@
+from ros_alarms import HandlerBase, AlarmBroadcaster
+from roboteq_msgs.msg import Status
+import rospy
+
+
+class ThrusterFault(HandlerBase):
+    alarm_name = 'thruster-fault'
+
+    def __init__(self):
+        self.broadcaster = AlarmBroadcaster(self.alarm_name)
+
+        # Thruster topics to listen to
+        motor_topics = ['/FL_motor', '/FR_motor', '/BL_motor', '/BR_motor']
+
+        # Subscribe to the status message for all thruster topics
+        [rospy.Subscriber(
+            topic + '/status', Status, self._check_faults, topic) for topic in motor_topics]
+
+        # Dictionary for the faults as defined in roboteq_msgs/Status
+        self.fault_codes = {1: 'OVERHEAT', 2: 'OVERVOLTAGE', 4: 'UNDERVOLTAGE', 8: 'SHORT_CIRCUIT',
+                            16: 'EMERGENCY_STOP', 32: 'SEPEX_EXCITATION_FAULT', 64: 'MOSFET_FAILURE',
+                            128: 'STARTUP_CONFIG_FAULT'}
+
+        self._raised = False
+
+    def _check_faults(self, msg, topic):
+        if msg.fault == 0:
+            return
+
+        # Figure out what fault codes have been raised and find their meaning from dictionary
+        get_fault_codes = []
+        for key, value in self.fault_codes.iteritems():
+            # Fault message is the sum of binary strings
+            decode = msg.fault & key
+            if decode != 0:
+                get_fault_codes.append(value)
+
+        if not self._raised:
+            self.broadcaster.raise_alarm(
+                severity=5,
+                problem_description='{} has faults: {}'.format(
+                    topic, ' '.join(get_fault_codes)),
+                parameters={'fault': msg.fault}
```

Comment:
Should report if multiple thrusters have faults, so needs to internally track each thrusters fault then raise alarm if there are any changes. The list of faults by thruster should be in the parameters so other programs can decode them as json instead of in the problem description string